plugins/outputs/prometheus_client/v1/collector.go

```diff
@@ -212,8 +218,6 @@ func sorted(metrics []telegraf.Metric) []telegraf.Metric {
 
 func (c *Collector) Add(metrics []telegraf.Metric) error {
 	c.Lock()
-	defer c.Unlock()
-
```

Comment:
Moving this to an explicit call is fragile and might pose a foot-gun for future changes (e.g. if someone decides to return in that long function). Would moving the locked-part to an own function with the deferrer be an option?

Reply:
I was having doubts about this as well, but didn't comment about it because it currently doesn't have a return.